discovery.rst

```diff
@@ -22,36 +16,63 @@ Currently available discovery services:
 
 The principle is always the same: you call the static ``find`` method on the discovery service if no explicit
 implementation was specified. The discovery service will try to locate a suitable implementation.
-If no implementation is found, an ``Http\Discovery\NotFoundException`` is thrown.
+If no implementation is found, an ``Http\Discovery\Exception\NotFoundException`` is thrown.
+
+.. versionadded:: 0.9
+    The exception ``Http\Discovery\NotFoundException`` has been moved to ``Http\Discovery\Exception\NotFoundException``.
+
+Discovery is simply a convenience wrapper to statically access clients and factories for when
+Dependency Injection is not an option. Discovery is useful in libraries that want to offer
+zero-configuration services relying on the virtual packages.
+
+
+Strategies
+----------
+
+The package supports multiple discovery strategies and comes with two out-of-the-box:
+
+- A built-in strategy with official HTTPlug components
+- A `Puli`_ strategy
+
+Strategies provide candidates of a type which gets evaluated by the discovery service.
+When it finds the best candidate, it caches it and stops looking in further strategies.
+
 
 Installation
 ------------
 
+.. code-block:: bash
+
+    $ composer require php-http/discovery
+
+
+Using Puli
+^^^^^^^^^^
+
 There are two kinds of installation:
 
 - In a reusable library
 - In an application
 
-In both cases you have to install the discovery package itself:
-
-.. code-block:: bash
-
-    $ composer require php-http/discovery
+In both cases you have to install the discovery package itself and set up Puli.
```

Comment:
should a reusable library ever care about puli now? i guess not - except if it wants if for ci testing, but that feels like quite an edge case. the library can depend on discovery alone now. only when using the library with discovery and wanting puli should you have to care about it.